### PR TITLE
adds support for ubuntu 2204 in image-builder

### PR DESCRIPTION
**Pull request by jaxesn**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/aws/image-builder/builder/builder.go [hunk 1]
```diff
@@ -115,43 +114,19 @@ func (b *BuildOptions) BuildImage() {
 			}
 		}
 
-		if b.Os == Ubuntu {
```
Comment (jaxesn) [inline projects/aws/image-builder/builder/builder.go:118]: I think we should actually check the existence of the `/usr/share/edk2/ovmf/OVMF_CODE.fd` and the `OVMF.fd` file and depending on which exists, make sure our image-builder code is passing that. this way al2 and ubuntu/rhel hosts should work for this process. thoughts?

File: projects/kubernetes-sigs/image-builder/Makefile [hunk 2]
```diff
@@ -153,231 +186,262 @@ EXCLUDE_FROM_CHECKSUMS_BUILDSPEC=true
...
-	@echo -e $(call TARGET_END_LOG)
+##############################################################
 
-$(FINAL_BOTTLEROCKET_AMI_PATH): FULL_OUTPUT_DIR=$(MAKE_ROOT)/$(OUTPUT_DIR)/bottlerocket
-$(FINAL_BOTTLEROCKET_AMI_PATH): BOTTLEROCKET_DOWNLOAD_PATH=$(FULL_OUTPUT_DIR)/bottlerocket/downloads
-$(FINAL_BOTTLEROCKET_AMI_PATH):
+########################### FAKE TARGETS ###############################
+# Since we do not build the ova in presubmit but want to validate upload-artifacts behavior
+# ex: fake-ubuntu-2004-raw.gz
+fake-%: PAKER_LOG_PATH=$(ARTIFACTS_PATH)/$(subst -,/,$(basename $(*F)))/packer.log
```
Comment (abhay-krishna) [inline projects/kubernetes-sigs/image-builder/Makefile:215]:
```suggestion
fake-%: PACKER_LOG_PATH=$(ARTIFACTS_PATH)/$(subst -,/,$(basename $(*F)))/packer.log
```

File: projects/kubernetes-sigs/image-builder/Makefile [hunk 3]
```diff
@@ -153,231 +186,262 @@ EXCLUDE_FROM_CHECKSUMS_BUILDSPEC=true
...
+fake-%:
 	@echo -e $(call TARGET_START_LOG)
-	mkdir -p $(@D)
-	mv $(BOTTLEROCKET_DOWNLOAD_PATH)/ami/*.gz $@
+	touch $(@)
+	touch $(PAKER_LOG_PATH)
```
Comment (abhay-krishna) [inline projects/kubernetes-sigs/image-builder/Makefile:219]: nit:
```suggestion
	touch $(PACKER_LOG_PATH)
```

File: projects/kubernetes-sigs/image-builder/build/setup_packer_configs.sh [hunk 4]
```diff
@@ -134,29 +135,37 @@ env_and_envsubst '$EKSD_NAME' \
...
-    cat <<< $(jq --arg httpendpoint "http://$PACKER_HTTP_SERVER_IP:{{ .HTTPPort }}" \
-             '.boot_media_path=$httpendpoint' $rhel_ova_config_file) > $rhel_ova_config_file
+
+    PACKER_HTTP_SERVER_IP=${PACKER_HTTP_SERVER_IP:-$HTTP_SERVER_IP}
+    if [ -z $PACKER_HTTP_SERVER_IP ]; then
+        echo "PACKER_HTTP_SERVER_IP cannot be automatically determined.  Please export PACKER_HTTP_SERVER_IP=<Current Host's IP>"
+        exit 1
+    fi
+
+    find ${MAKE_ROOT}/${IMAGE_BUILDER_DIR}/packer/ova -type f -name '*.json' | xargs sed -i "s/{{ .HTTPIP }}/$PACKER_HTTP_SERVER_IP/g"
```
Comment (abhay-krishna) [inline projects/kubernetes-sigs/image-builder/build/setup_packer_configs.sh:168]: Need this check to use the proper sed executable for Darwin
```suggestion
    SED=sed
    if [ "$(uname -s)" = "Darwin" ]; then
      SED=gsed
    fi
    find ${MAKE_ROOT}/${IMAGE_BUILDER_DIR}/packer/ova -type f -name '*.json' | xargs $SED -i "s/{{ .HTTPIP }}/$PACKER_HTTP_SERVER_IP/g"
```

**Pull request merged**